flake.nix

```diff
@@ -0,0 +1,124 @@
+{
+  description = "Nix package for centrifuge-chain";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-21.11;
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = github:nix-community/fenix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+  };
+
+  outputs = inputs:
+    let
+      name = "centrifuge-chain";
+      major = "0.10.9";
+      version = "${major}-${commit-substr}";
+
+      # This evaluates to the first 6 digits of the git hash of this repo's HEAD
+      # commit, or to "dirty" if there are uncommitted changes.
+      commit-substr = builtins.substring 0 6 (inputs.self.rev or "dirty");
+
+      # This could be made into a list, to support multiple platforms
+      system = "x86_64-linux";
+
+      pkgs = inputs.nixpkgs.legacyPackages.${system};
+
+      nightlyRustPlatform = pkgs.makeRustPlatform {
```

Comment:
As discussed offline, we may need a session around using specific rust toolchains and update of dependencies. AKA, how to make a link between `nightly-2021-11-07` and the proper version of the nix dependency.